lld/ELF/Arch/AMDGPU.cpp

```diff
@@ -143,7 +143,7 @@ uint32_t AMDGPU::calcEFlags() const {
   case ELFABIVERSION_AMDGPU_HSA_V6:
     return calcEFlagsV6();
   default:
-    Err(ctx) << "unknown abi version: " << abiVersion;
+    Err(ctx) << "unknown amd-hsa abi version: " << static_cast<int>(abiVersion);
```

Comment:
Needs a test. Perhaps Update  amdgpu-abi-version-err.s

I don't think the cast is necessary?

Usually "ABI" instead of abi. Why is amd-hsa in lower case?

Reply:
and why is it split with a -

Reply:
>I don't think the cast is necessary?

the cast is necessary because otherwise you get ASCII characters instead of the number itself on platforms where `sizeof(uint8_t) == sizeof(char)` (see eg [SO](https://stackoverflow.com/a/19562163/9045206)).

Reply:
> and why is it split with a -

I just matched what we use for the triple - doesn't matter to me - will change to `AMDHSA ABI`

Reply:
@MaskRay `amdgpu-abi-version-err.s` is not the right place because `llvm-mc` already [catches illegal ABI](https://github.com/makslevental/llvm-project/blob/3e742b517a0606bfed329dfcb8c34c614ed73ea7/llvm/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.cpp#L234) -  you need an actual ELF file. I don't see any such files around...

Reply:
Then use yaml2obj. lld/test/ELF has quite a few examples

Reply:
add amdgpu-abi-version-err-2.yml